server/models/Collective.js

```diff
@@ -858,6 +858,61 @@ Collective.prototype.getNextGoal = async function (until) {
   return nextGoal;
 };
 
+Collective.prototype.updateSocialLinks = async function (socialLinks) {
+  if (socialLinks.length > 10) {
+    throw new Error('account cannot set more than 10 social links');
+  }
+
+  if (socialLinks.length === 0) {
+    await models.SocialLink.destroy({
+      where: {
+        CollectiveId: this.id,
+      },
+    });
+    return [];
+  }
+
+  return await sequelize.transaction(async transaction => {
+    const existingLinks = await models.SocialLink.findAll({
+      where: {
+        CollectiveId: this.id,
+      },
+      transaction,
+      lock: true,
+    });
+
+    const removedLinks = differenceBy(existingLinks, socialLinks, sl => sl.url + sl.type);
+
+    if (removedLinks.length !== 0) {
+      await models.SocialLink.destroy({
+        where: {
+          CollectiveId: this.id,
+          [Op.and]: {
+            [Op.or]: removedLinks.map(rl => ({
+              url: rl.url,
+              type: rl.type,
+            })),
+          },
+        },
+        transaction,
+      });
+    }
+
+    return await models.SocialLink.bulkCreate(
```

Comment:
Could be interesting to record an activity to log this; unless another activity already covers it?

Reply:
Its part of the COLLECTIVE_EDITED activity

Reply:
Would be great to include it in `previousData`/`newData` then (see `getCollectiveDataDiff`). Doesn't have to be in this PR, we can followup in a separate issue.